crates/context/src/lib.rs

```diff
@@ -158,6 +158,7 @@ impl ContextManager {
         initialization_params: Vec<u8>,
         result_sender: oneshot::Sender<EyreResult<(ContextId, PublicKey)>>,
     ) -> EyreResult<()> {
+        let protocol_clone = protocol.to_owned();
```

Comment:
how about you take in a `String` instead of a `&str`

Reply:
So, this is unresolved. What I meant here is instead of the parameter `protocol` being a `&str`, now that we know we need a `String`, you can simply change that there to a `String`.

With that, you'd no longer need a `protocol_clone`.

Reply:
Line 154

>         protocol: &str,

let's make that `protocol: String`, at which point you can simplify the rest of the changes here and calling `query` / `mutate` should be `protocol.as_str().into()`

no `clone()`s necessary